plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/element/ApplyStaticNatVspCommand.java

```diff
@@ -51,13 +61,91 @@ public boolean isL3Network() {
         return _isL3Network;
     }
 
+    public boolean isVpc() {
+        return _isVpc;
+    }
+
     public List<Map<String, Object>> getStaticNatDetails() {
         return _staticNatDetails;
     }
 
+    public static class Builder implements CmdBuilder<ApplyStaticNatVspCommand> {
+        private String _networkDomainUuid;
+        private String _networkUuid;
+        private String _vpcOrSubnetUuid;
+        private boolean _isL3Network;
+        private boolean _isVpc;
+        private List<Map<String, Object>> _staticNatDetails;
+
+        public Builder networkDomainUuid(String networkDomainUuid) {
+            this._networkDomainUuid = networkDomainUuid;
+            return this;
+        }
+
+        public Builder networkUuid(String networkUuid) {
+            this._networkUuid = networkUuid;
+            return this;
+        }
+
+        public Builder vpcOrSubnetUuid(String vpcOrSubnetUuid) {
+            this._vpcOrSubnetUuid = vpcOrSubnetUuid;
+            return this;
+        }
+
+        public Builder isL3Network(boolean isL3Network) {
+            this._isL3Network = isL3Network;
+            return this;
+        }
+
+        public Builder isVpc(boolean isVpc) {
+            this._isVpc = isVpc;
+            return this;
+        }
+
+        public Builder staticNatDetails(List<Map<String, Object>> staticNatDetails) {
+            this._staticNatDetails = staticNatDetails;
+            return this;
+        }
+
+        @Override
+        public ApplyStaticNatVspCommand build() {
+            return new ApplyStaticNatVspCommand(_networkDomainUuid, _networkUuid, _vpcOrSubnetUuid, _isL3Network, _isVpc, _staticNatDetails);
+        }
+    }
+
     @Override
     public boolean executeInSequence() {
         return false;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        ApplyStaticNatVspCommand that = (ApplyStaticNatVspCommand) o;
+
+        if (_isL3Network != that._isL3Network) return false;
+        if (_isVpc != that._isVpc) return false;
+        if (_networkDomainUuid != null ? !_networkDomainUuid.equals(that._networkDomainUuid) : that._networkDomainUuid != null)
+            return false;
+        if (_networkUuid != null ? !_networkUuid.equals(that._networkUuid) : that._networkUuid != null) return false;
+        if (_staticNatDetails != null ? !_staticNatDetails.equals(that._staticNatDetails) : that._staticNatDetails != null)
+            return false;
+        if (_vpcOrSubnetUuid != null ? !_vpcOrSubnetUuid.equals(that._vpcOrSubnetUuid) : that._vpcOrSubnetUuid != null)
+            return false;
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
This ``hashCode`` implementation should include attributes from the ``Command`` super class.